From db187e95f62d9bff268e68f39b441b6da8be9111 Mon Sep 17 00:00:00 2001
From: Michael Simons <msimons@microsoft.com>
Date: Mon, 2 Aug 2021 19:02:27 +0000
Subject: [PATCH] Run GenerateFiles during source build

GenerateFiles is not running in source build yet the outputs are required.
These changes get GenerateFiles to run as part of source build.
These changes also turn on the binaryLog option while building the RepoTasks during source build.
---
 eng/SourceBuild.props                        | 3 +--
 eng/Tools.props                              | 2 +-
 eng/tools/GenerateFiles/GenerateFiles.csproj | 1 +
 4 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/eng/SourceBuild.props b/eng/SourceBuild.props
index 27cbe92036..3dbfe2ce7e 100644
--- a/eng/SourceBuild.props
+++ b/eng/SourceBuild.props
@@ -1,5 +1,4 @@
 <Project>
-
   <PropertyGroup>
     <GitHubRepositoryName>aspnetcore</GitHubRepositoryName>
     <SourceBuildManagedOnly>true</SourceBuildManagedOnly>
@@ -47,7 +46,7 @@
           BeforeTargets="Execute">
 
     <Exec
-      Command="./eng/build.sh --only-build-repo-tasks"
+      Command="./eng/build.sh --only-build-repo-tasks -bl"
       WorkingDirectory="$(InnerSourceBuildRepoRoot)"
       EnvironmentVariables="@(InnerBuildEnv)" />
   </Target>
diff --git a/eng/Tools.props b/eng/Tools.props
index a285c2406a..a38257f770 100644
--- a/eng/Tools.props
+++ b/eng/Tools.props
@@ -13,7 +13,7 @@
   <!-- Update the generated files when we restore projects. Skip in desktop msbuild due to VS 16.8 requirements. -->
   <Target Name="GenerateDirectoryBuildFiles"
       AfterTargets="Restore"
-      Condition=" '$(DotNetBuildFromSource)' != 'true' AND '$(MSBuildRuntimeType)' == 'core' ">
+      Condition=" '$(MSBuildRuntimeType)' == 'core' ">
     <!-- Separate invocations and use different properties to ensure second can load the restored package info. -->
     <MSBuild Projects="$(RepoRoot)eng\tools\GenerateFiles\GenerateFiles.csproj"
         RemoveProperties="BaseIntermediateOutputPath"
diff --git a/eng/tools/GenerateFiles/GenerateFiles.csproj b/eng/tools/GenerateFiles/GenerateFiles.csproj
index 0ae5c9753e..dbf9965bf9 100644
--- a/eng/tools/GenerateFiles/GenerateFiles.csproj
+++ b/eng/tools/GenerateFiles/GenerateFiles.csproj
@@ -2,6 +2,7 @@
   <PropertyGroup>
     <!-- Use fixed version instead of $(DefaultNetCoreTargetFramework) to avoid needing workarounds set up here. -->
     <TargetFramework>net5.0</TargetFramework>
+    <ExcludeFromSourceBuild>false</ExcludeFromSourceBuild>
   </PropertyGroup>
 
   <ItemGroup>
-- 
2.29.2

